### DISCUSSION TITLE
RabbitMQOperator task stuck on queued when using CeleryExecutor as core executor

### DISCUSSION THREAD

**Discussion by abdalhalimalzohbi** [apache/airflow#43351] [General] [answered]
### Apache Airflow Provider(s)

redis

### Versions of Apache Airflow Providers

airflow-provider-rabbitmq 0.6.1

### Apache Airflow version

2.10.2

### Operating System

airflow bitnami image

### Deployment

Official Apache Airflow Helm Chart

### What happened

When using CeleryExecutor as the core executor for Airflow and utilizing the RabbitMQOperator task from rabbitmq_provider.operators.rabbitmq, the task remains stuck in the queued state and is not recognized by Celery workers. The issue arises when attempting to use RabbitMQOperator to publish messages to an external RabbitMQ queue, which is not related to the Airflow instance.

The root cause of the conflict is that the RabbitMQOperator takes a queue field as a parameter, and this value is mistakenly treated as the name of the queue that the Celery workers will use to execute the task. This creates confusion, as the queue parameter is intended to represent the destination queue for message publishing, not the execution queue for Celery tasks.

### How to reproduce

NA

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by eladkal** [comment, marked as answer]
`airflow-provider-rabbitmq` is not official Airflow provider. There is no `RabbitMQOperator` in the Apache-Airflow project.
Unless you can explain what is the bug in Airflow code base I suggest to submit this report to the maintainers of `airflow-provider-rabbitmq`

**Reply by abdalhalimalzohbi** [reply]
Thank you so much, when using the sequential executor, the operator publishes to the correct external queue and does not get stuck. however with the celery executor, it gets stuck because it mistakenly assigns the queue name as the task executor, which caused the confusion. i removed the airflow-provider-rabbitmq package, created a custom operator, and used pika to handle my use case.

**Comment by potiuk** [comment]
Yep., This question is to whoever provides the operator - not here. Converted to discussion
